src/include/miopen/batchnorm/solvers.hpp

```diff
@@ -132,34 +137,210 @@ struct BnFwdInference final : BatchnormSolver
                              const miopen::batchnorm::ProblemDescription& problem) const override;
 };
 
-struct BnCKFwdInference final : BatchnormSolver
+struct PerformanceConfigBnCKFwdInference : PerfConfigBase<PerformanceConfigBnCKFwdInference>
+{
+    int index;
+    std::string kernel_id;
+    std::vector<std::string> valid_kernels;
+    PerformanceConfigBnCKFwdInference(int idx, std::string kernl_id)
+        : index(idx), kernel_id(kernl_id)
+    {
+    }
+    PerformanceConfigBnCKFwdInference() : PerformanceConfigBnCKFwdInference(0, "") {}
+    PerformanceConfigBnCKFwdInference(bool) : PerformanceConfigBnCKFwdInference(0, "") {}
+    MIOPEN_INTERNALS_EXPORT void
+    HeuristicInit(const miopen::batchnorm::ProblemDescription& problem_desc);
+    MIOPEN_INTERNALS_EXPORT bool
+    SetNextValue(const miopen::batchnorm::ProblemDescription& problem_desc);
+    MIOPEN_INTERNALS_EXPORT bool IsValidValue() const;

```

Comment:
It seems unused.

Also what kind of value it validates?

Reply:
Unless something changed, it is required by `GenericSearch`. Idea is that search gets initial config and then steps through them, skipping ones that are not expected to produce a working kernel, which is reported by this function.